lighthouse-core/report/report-generator.js

```diff
@@ -202,10 +202,22 @@ class ReportGenerator {
 
   /**
    * Gets the script for the report UI
-   * @return {string}
+   * @param {string} reportContext
+   * @return {Array<string>}
    */
-  getReportJS() {
-    return fs.readFileSync(path.join(__dirname, './scripts/lighthouse-report.js'), 'utf8');
+  getReportJS(reportContext) {
+    const scriptList = [];
+
+    if (reportContext === 'devtools') {
+      return scriptList;
+    }
+
+    if (reportContext === 'perf-x') {
+      scriptList.push(fs.readFileSync(path.join(__dirname, './scripts/perf-x-api.js')));
+    }
+
+    scriptList.push(fs.readFileSync(path.join(__dirname, './scripts/lighthouse-report.js')));
```

Comment:
awkward logic here.
probably better to do a `reportContext !== 'devtools' conditional around this last push(), and nuke the early exit on 211